src/m365/spo/commands/commandset/commandset-set.ts

```diff
@@ -0,0 +1,225 @@
+      },
+      {
+        option: '--newTitle [newTitle]'
+      },
+      {
+        option: '-l, --listType [listType]', autocomplete: SpoCommandsetSetCommand.listTypes
+      },
+      {
+        option: '--clientSideComponentProperties  [clientSideComponentProperties]'
+      },
+      {
+        option: '-s, --scope [scope]', autocomplete: SpoCommandsetSetCommand.scopes
+      },
+      {
+        option: '--location [location]', autocomplete: SpoCommandsetSetCommand.locations
+      }
+    );
+  }
+
+  #initValidators(): void {
+    this.validators.push(
+      async (args: CommandArgs) => {
+        if (args.options.id && !validation.isValidGuid(args.options.id as string)) {
+          return `${args.options.id} is not a valid GUID`;
+        }
+
+        if (args.options.clientSideComponentId && !validation.isValidGuid(args.options.clientSideComponentId as string)) {
+          return `${args.options.clientSideComponentId} is not a valid GUID`;
+        }
+
+        if (args.options.listType && SpoCommandsetSetCommand.listTypes.indexOf(args.options.listType) < 0) {
+          return `${args.options.listType} is not a valid list type. Allowed values are ${SpoCommandsetSetCommand.listTypes.join(', ')}`;
+        }
+
+        if (args.options.scope && SpoCommandsetSetCommand.scopes.indexOf(args.options.scope) < 0) {
+          return `${args.options.scope} is not a valid scope. Allowed values are ${SpoCommandsetSetCommand.scopes.join(', ')}`;
+        }
+
+        if (args.options.location && SpoCommandsetSetCommand.locations.indexOf(args.options.location) < 0) {
+          return `${args.options.location} is not a valid location. Allowed values are ${SpoCommandsetSetCommand.locations.join(', ')}`;
+        }
+
+        if (!args.options.newTitle && !args.options.listType && !args.options.clientSideComponentProperties && !args.options.location) {
+          return `Please specify option to be updated`;
+        }
+
+        return validation.isValidSharePointUrl(args.options.webUrl);
+      }
+    );
+  }
+
+  #initOptionSets(): void {
+    this.optionSets.push(
+      { options: ['id', 'title', 'clientSideComponentId'] }
+    );
+  }
+
+  public async commandAction(logger: Logger, args: CommandArgs): Promise<void> {
+    if (this.verbose) {
+      logger.logToStderr(`Updating ListView Command Set ${args.options.clientSideComponentId} to site '${args.options.webUrl}'...`);
+    }
+
+    if (!args.options.scope) {
+      args.options.scope = 'Site';
+    }
+
+    const location: string = this.getLocation(args.options.location ? args.options.location : '');
+
+    try {
+      const requestBody: any = {
+      };
+
+      if (args.options.newTitle) {
+        requestBody.Title = args.options.newTitle;
+      }
+
+      if (args.options.location) {
+        requestBody.Location = location;
+      }
+
+      if (args.options.listType) {
+        requestBody.RegistrationId = args.options.listType === 'List' ? "100" : "101";
```

Comment:
like in the previous comment I was thinking we could extend to include SitePages lib here as well